### ISSUE TITLE
Warning S2583 Change this condition so that it does not always evaluate to 'false'; some subsequent code is never executed.

### ISSUE THREAD

**Issue by Peska** [SonarSource/sonarlint-visualstudio#866]
### Description

Warning S2583 Change this condition so that it does not always evaluate to 'false'; some subsequent code is never executed.

### Repro steps

Sample class to reproduce. Warning in line 5:

```
public class TestClass
{
	public void TestMethod(bool? input)
	{
		if (input ?? false)
			Console.WriteLine("input is true");
		else
			Console.WriteLine("input is false");
	}
}
```

### Expected behavior

No warning given. This is a false positive.

### Actual behavior

Warning S2583 Change this condition so that it does not always evaluate to 'false'; some subsequent code is never executed.

### Known workarounds

Use:

`if (input.GetValueOrDefault())`

### Related information

* SonarLint for VS Version: 4.9.0.4314 https://github.com/SonarSource/sonarlint-visualstudio/releases/tag/v4.9-rc2
* Visual Studio Version Enterprise 2019 16.0.0
* SonarQube server version: ---
* Languages in your solution: C#

**Comment by duncanp-sonar** [comment]
@Peska thanks for reporting this issue.
FYI the analyzer rules are in the [sonar-dotnet repo](https://github.com/SonarSource/sonar-dotnet). I've created a [copy of this issue](https://github.com/SonarSource/sonar-dotnet/issues/2369) there, and I'll close this one.
